### DISCUSSION TITLE
Reverse Proxy redirecting to forward proxy server

### DISCUSSION THREAD

**Discussion by Opienm** [dotnet/yarp#960] [Q&A] [answered]
Hello,
I have been trying to find a solution for several months without success..
My goal :
I need to be able to decrypt the https client requests without the client having an ssl certificate installed. So i think that  **i need to use a reverse proxy that will process the request using the proxy server ssl certificate, and then send the response back to the client**.

``` 
+---------+    Request    +-----------+                 +------------+                 +-------+
|         +-------------> |           |     Request     |            |     Request     |       |
|         |               |  Reverse  +---------------> |   Proxy    |---------------> |  Web  |
|         |               |   Proxy   |                 |   Server   |                 |       |
| Client  |               |           |                 |            |                 |       |
|         |               |           |    Response     |            |    Response     |       |
|         |    Response   |           | <---------------+            | <---------------+       |
|         | <-------------+           |                 |            |                 |       |
+---------+               +-----------+                 +------------+                 +-------+
```

I think that this reverse proxy does not support the CONNECT request that forwards proxies use. So want i want to achieve is impossible with microsoft reverse-proxy.

Could someone help me with what I could use as a reverse proxy to solve my problem? 
Thanks

**Comment by Tratcher** [comment, marked as answer]
TLS and CONNECT were specifically designed to prevent this man-in-the-middle decryption attack. A decrypting proxy MUST use a trusted cert installed on the client.

**Reply by Opienm** [reply]
Even by using a transparent proxy / reverse proxy before it goes through to the actual proxy server ?

**Reply by Tratcher** [reply]
Yes. The client must trust the proxy.
